components/compliance-service/api/tests/66_A2V6_migration_spec.rb

```diff
@@ -0,0 +1,343 @@
+    assert_equal(4, res['profiles'][0]['controls'].length)
+
+    control_ids = res['profiles'][0]['controls'].map {|c| c.id}
+    assert_equal(["nginx-01", "nginx-02", "nginx-03", "nginx-04"], control_ids)
+
+    assert_equal(Reporting::Report, res.class)
+
+    assert_equal('nginx-01', res['profiles'][0]['controls'][0]['id'])
+    assert_equal(1, res['profiles'][0]['controls'][0]['impact'])
+    assert_equal(Google::Protobuf::Map, res['profiles'][0]['controls'][0]['tags'].class)
+    assert_equal('Running worker process as non-privileged user', res['profiles'][0]['controls'][0]['title'])
+    assert_equal(1, res['profiles'][0]['controls'][0]['results'].length)
+
+    passed_control = res['profiles'][0]['controls'][0]['results'].first
+    assert_equal('passed', passed_control['status'])
+
+    assert_equal('failed', res['profiles'][1]['status'])
+    assert_equal('', res['profiles'][1]['skip_message'])
+  end
+
+  it "control list itemsfor 2018-02-09 with a size of 3" do
+    actual_data = GRPC reporting, :list_control_items, Reporting::ControlItemRequest.new(
+        filters: [
+            Reporting::ListFilter.new(type: 'start_time', values: ['2018-02-09T00:00:00Z']),
+            Reporting::ListFilter.new(type: 'end_time', values: ['2018-02-09T23:59:59Z'])
+        ],
+        size: 3
+    )
+
+    expected_data =
+    { "controlItems"=>
+      [{"id"=>"apache-01",
+        "title"=>"Apache should be running",
+        "profile"=>
+         {"title"=>"DevSec Apache Baseline",
+          "id"=>"41a02784bfea15592ba2748d55927d8d1f9da205816ef18d3bb2ebe4c5ce18a8",
+          "version"=>"2.0.0"},
+        "impact"=>1,
+        "endTime"=>"2018-02-09T09:18:41Z",
+        "controlSummary"=>
+         {"total"=>1,
+          "passed"=>{},
+          "skipped"=>{"total"=>1},
+          "failed"=>{},
+          "waived"=>{}}},
+       {"id"=>"apache-02",
+        "title"=>"Apache should be enabled",
+        "profile"=>
+         {"title"=>"DevSec Apache Baseline",
+          "id"=>"41a02784bfea15592ba2748d55927d8d1f9da205816ef18d3bb2ebe4c5ce18a8",
+          "version"=>"2.0.0"},
+        "impact"=>1,
+        "endTime"=>"2018-02-09T09:18:41Z",
+        "controlSummary"=>
+         {"total"=>1,
+          "passed"=>{},
+          "skipped"=>{"total"=>1},
+          "failed"=>{},
+          "waived"=>{}}},
+       {"id"=>"apache-03",
+        "title"=>"Apache should start max. 1 root-task different",
+        "profile"=>
+         {"title"=>"DevSec Apache Baseline",
+          "id"=>"41a02784bfea15592ba2748d55927d8d1f9da205816ef18d3bb2ebe4c5ce18a8",
+          "version"=>"2.0.0"},
+        "impact"=>0.5,
+        "endTime"=>"2018-02-09T09:18:41Z",
+        "controlSummary"=>
+         {"total"=>1,
+          "passed"=>{"total"=>1},
+          "skipped"=>{},
+          "failed"=>{},
+          "waived"=>{}}}]
+    }
+    control_item_array = actual_data['control_items']
+    assert_equal_json_content(expected_data, actual_data)
```

Comment:
This comment applies here as well: https://github.com/chef/automate/pull/2707/files#r374922954